src/liteclient.cc

```diff
@@ -261,6 +263,49 @@ void LiteClient::notify(const Uptane::Target& t, std::unique_ptr<ReportEvent> ev
   }
 }
 
+void LiteClient::importRootMetaIfNeededAndPresent() {
+  std::string data;
+  if (storage->loadRoot(&data, Uptane::RepositoryType::Image(), Uptane::Version(2))) {
+    LOG_DEBUG << "Root role metadata are already present in DB, skip their import from FS";
+    return;
+  }
+  boost::filesystem::path tuf_meta_src{config.storage.uptane_metadata_path.get("/")};
+  if (tuf_meta_src == "/metadata") {
+    // if libaktualizr's default value then set aklite's default value
+    tuf_meta_src = "/usr/lib/sota/tuf";
+  }
+  if (!boost::filesystem::exists(tuf_meta_src)) {
+    LOG_INFO << "Root metadata are not provisioned to FS, proceed with fetching them from Device Gateway considering "
+                "Trust On First Use";
+    return;
+  }
+
+  const std::string prod_bc_value{"production"};
```

Comment:
how does this work for offline updates where the device doesn't have a certificate?

Reply:
The cert is needed solely to determine a device type here, so we know which of the root meta set to import, ci or prod. In the case of an offline update a device just fetches what it was pointed to.

Reply:
> In the case of an offline update a device just fetches what it was pointed to.

Can you elaborate here, I still don't understand. It looks like for an offline update, there will be no client cert, the LiteClient constructor will call this method, the `key_manager->getBC` will throw an error, we'll log it and then wind up defaulting to prod=false.

Reply:
So, no certs means that this is CI device.

Reply:
i believe the current offline updates requirements doc says devices will not be registered.

Reply:
I thought it makes sense only for dev devices, let's discuss it offline today.

Reply:
how about you default to production *if* the production directory has content? Then people only doing CI targets work by default, but people that have production will do the best option by default.

Reply:
I changed it to the two-level check as we discussed offline.

Reply:
I resolve this conversation since I reworked implementation considerably.